src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/variabledeclarationusagedistance/InputVariableDeclarationUsageDistanceGeneral2.java

```diff
@@ -63,4 +63,52 @@ void method() throws Exception {
             a.equals("");
         }
     }
+
+    void method2() {
+        int a = 12; // violation
+        if (true) {
+            method2();
+            p();
+            method2();
+            a++;
+        }
+    }
+
+    void checkIfStatementWithoutParen() {
+        int a = 12; // violation
+        method2();
+        if (true)
+            a++;
+
+        int b = 12; // violation
+        method2();
+        if (false)
+            method2();
+        else if(true)
+            b++;
+
+        int c = 12; // violation
+        method2();
+        p();
+        if (true)
+            c++;
+        else
+            method2();
+    }
+
+    void testConsecutiveIfStatements() {
+        int a = 12; // ok
+        int b = 13; // violation
+        int c = 14; // violation
+        int d = 15; // violation
+
+        if (true)
+            a++;
+        if (true)
+            b++;
+        if (false)
+            c++;
+        if (true)
+            d++;
+    }
```

Comment:
When `return` is the first statement in `if` the last child is not `SEMI`:
```java
if (true) return a;
```
```
LITERAL_IF -> if [3:8]
|--LPAREN -> ( [3:11]
|--EXPR -> EXPR [3:12]
|  `--LITERAL_TRUE -> true [3:12]
|--RPAREN -> ) [3:16]
`--LITERAL_RETURN -> return [3:18]
    |--EXPR -> EXPR [3:25]
    |  `--IDENT -> a [3:25]
    `--SEMI -> ; [3:26]
```
And it doesn't catch a violation in such code:
```java
    String testReturn() {
        int a = 1; // violation
        System.out.println();
        System.out.println();
        System.out.println();
        if (true)
            return String.valueOf(a);

        return null;
    }
```
Can you please add it to the inputs too?

Reply:
I tested it, it won't give a violation, the `if` statement evaluations works correctly, but this is considered by check logic as an initialization sequence 
(`isInitializationSequence = true`), so it does not give a violation finally at:
https://github.com/checkstyle/checkstyle/blob/f4ab4aa20d27153bb9489315644cfb2a4c8b9e3c/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L417-L418

Reply:
I added a similar example as proof of working on current changes.